components/ui/back-to-top-button.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { animateScroll as scroll } from 'react-scroll';
+import { FaArrowUp } from 'react-icons/fa';
+import { Button } from './button';
+
+const BackToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const handleScroll = () => {
+    const yOffset = window.scrollY;
+
+    if (yOffset > 300) {
+      setIsVisible(true);
+    } else {
+      setIsVisible(false);
+    }
+  };
+
+  const handleScrollToTop = () => {
+    scroll.scrollToTop({ duration: 500, smooth: 'easeInOutQuad' });
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return (
+    <div
+      style={{
+        opacity: isVisible ? 1 : 0,
+        transition: 'opacity 0.3s',
+      }}
+      className="fixed right-[20px] bottom-[150px] md:bottom-[50px]"
+    >
+      <Button
+        variant="default"
+        onClick={handleScrollToTop}
+        className="h-[60px] w-[60px] rounded-full drop-shadow-lg"
+      >
+        <FaArrowUp size={48} />
```

Comment:
i'm not 100% fine with the size of the button/icon. In mobile it seems bigger than the existing FAB button of add project. also the style doesn't match with the transparent kind of theme. Let me know your thoughts. thanks